src/ext/Util/ca/scaexec.cpp

```diff
@@ -680,6 +707,34 @@ static HRESULT RemoveUserInternal(
     return hr;
 }
 
+static void GetServerName(LPWSTR pwzDomain, LPWSTR* ppwzServerName)
+{
+    PDOMAIN_CONTROLLER_INFOW pDomainControllerInfo = NULL;
+    UINT er;
+
+    if (pwzDomain && *pwzDomain)
+    {
+        er = ::DsGetDcNameW(NULL, (LPCWSTR)pwzDomain, NULL, NULL, NULL, &pDomainControllerInfo);
+        if (RPC_S_SERVER_UNAVAILABLE == er)
+        {
+            // MSDN says, if we get the above error code, try again with the "DS_FORCE_REDISCOVERY" flag
+            er = ::DsGetDcNameW(NULL, (LPCWSTR)pwzDomain, NULL, NULL, DS_FORCE_REDISCOVERY, &pDomainControllerInfo);
+        }
+        if (ERROR_SUCCESS == er)
+        {
+            *ppwzServerName = pDomainControllerInfo->DomainControllerName + 2;  //Add 2 so that we don't get the \\ prefix
```

Comment:
Should we verify that the domain `DomainControllerName` actually starts with `\\` before removing the first two characters? The documentation says the name is prefixes with `\` but I'm guessing that is a translation hiccup and they really meant `\\`. Still, it'd be bad to accidentally access dead memory here if the domain name was a single letter and missing the `\\`.

Reply:
Done.